File: Data Structures & Algorithms/n-queens/submission-0.cpp
class Solution {
public:
    bool underAttack(vector<string>& board, int n, int x, int y) {
        for(int i=1; i<n; i++) {
            if(y-i<0) break;
            if(board[y-i][x]=='Q') return true;
            if(x-i>=0 && board[y-i][x-i]=='Q') return true;
            if(x+i<n && board[y-i][x+i]=='Q') return true;
        }
        return false;
    }
    vector<vector<string>> recursive(vector<string> curr_board, int n, int col)
    {
        if(col>=n) return {curr_board};
        vector<vector<string>> res;
        for(int i=0; i<n; i++) {
            if(!underAttack(curr_board,n,i,col)) {
                auto newBoard = curr_board;
                newBoard[col][i]='Q';
                auto newRes = recursive(newBoard,n,col+1);
                res.insert(res.end(),newRes.begin(),newRes.end());
            }
        }
        return res;
    }

    vector<vector<string>> solveNQueens(int n) {
        string all_stars = "";
        for(int i=0; i<n; i++) all_stars+='.';
        vector<string> emptyBoard(n, all_stars);

        return recursive(emptyBoard,n,0);        
    }
};

File: Data Structures & Algorithms/valid-sudoku/submission-0.py
class Solution:
    def isValidSudoku(self, board: List[List[str]]) -> bool:
        for i in range(9):
            check = set()
            for j in range(9):
                if board[i][j]==".":
                    continue
                if board[i][j]<"1" or board[i][j]>"9" or (board[i][j] in check):
                    # print("1")
                    return False
                check.add(board[i][j])

        for j in range(9):
            check = set()
            for i in range(9):
                if board[i][j] == ".":
                    continue 
                if board[i][j] in check:
                    # print(2)
                    return False
                check.add(board[i][j])

        for si in range(0,9,3):
            for sj in range(0,9,3):
                check = set()
                for i in range(3):
                    for j in range(3):
                        if board[si+i][sj+j] == ".":
                            continue 
                        if board[si+i][sj+j] in check:
                            # print(3)
                            return False
                        check.add(board[si+i][sj+j])

        return True